src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (first == null) {
+            newNode = new Node<>(null, value, null);
+            first = newNode;
+        } else {
+            newNode = new Node<>(last, value, null);
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        Node<T> newNode;
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            newNode = new Node<>(null, value, first);
+            first.prev = newNode;
+            first = newNode;
+            size++;
+        } else {
+            Node<T> nodeByIndex = getNodeByIndex(index);
+            newNode = new Node<>(nodeByIndex.prev, value, nodeByIndex);
+            nodeByIndex.prev.next = newNode;
+            nodeByIndex.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> node = getNodeByIndex(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> deleteNode = getNodeByIndex(index);
```

Comment:
You don't need checkIndex() here, you already use it in getNodeByIndex().

Reply:
Done)